src/npi/__main__.py

```diff
@@ -1,31 +1,58 @@
 from argparse import ArgumentParser
+import pickle as pkl
+import torch
+from npi.config import NPIConfig
+from npi.dataset.npi_dataset import NPIDataSet
+from npi.models import NPITrainingModels, Classifier
 from npi.training.test_classifier import test_classifier
-from npi.training.test_npi import test_npi
+
+# from npi.training.test_npi import test_npi
 from npi.training.train_classifier import train_classifier
 
-from npi.training.train_npi import train_adversarial_NPI
+from npi.training import NPITrainer
+
 
 def train(args):
-    model = args["model"]
-    if(model == "npi"):
-        train_adversarial_NPI(args)
-    elif(model == "classifier"):
+    model = args.model
+    device = torch.device("cuda:0")  # TODO: Hyperparam
+    num_epochs = 6  # TODO: Hyperparam
+    split_ratio = 0.25  # TODO: Hyperparam
+    config = NPIConfig(device, save_folder="../models/npi_models/")
+    models = NPITrainingModels(
+        config,
+        content_classifier_path="../notebooks/politics/classifiers/layers_5_11/Classifier_classification_network_epoch0.bin",
+    )
+    if model == "npi":
+        trainer = NPITrainer(config, batch_size=4, headstart=0)
+        with open("../data/processed/politics/sentence_arrays.pkl_0", "rb") as datafile:
+            dataset = pkl.load(datafile)
+        max_train = len(dataset) - int(split_ratio * len(dataset))
+        train_data = NPIDataSet(dataset[:max_train], config, return_row=True)
+        test_data = NPIDataSet(dataset[max_train:], config, return_row=True)
+        trainer.train_adversarial_npi(
+            models, num_epochs, train_data, test_data
+        )
+    elif model == "classifier":
         train_classifier(args)
     else:
         print("Only can train classifier or npi.")
 
+
 def test(args):
     model = args["model"]
-    if(model == "npi"):
-        test_npi(args)
-    elif(model == "classifier"):
+    if model == "npi":
+        # test_npi(args)
+        pass
+    elif model == "classifier":
         test_classifier(args)
     else:
         print("Only can train classifier or npi.")
 
+
 def construct(args):
```

Comment:
Trevor, if I recall, you were working on construct_data right? You can add code in here to get construct working. Don't worry about the args for now, just hard code the parameters that work into a variable and we can work from there.